q2cli/builtin/tools.py

```diff
@@ -195,7 +196,104 @@ def peek(path):
         click.echo(metadata.format)
 
 
-@tools.command('inspect-metadata',
+@tools.command(name='cast-metadata',
+               short_help='Designate metadata column types.',
+               help='Designate metadata column types.'
+                    ' Supported column types can be found on docs.qiime2.org.'
+                    ' Providing multiple file paths to this command will merge'
+                    ' the metadata.',
+               cls=ToolCommand)
+@click.option('--cast', required=True, metavar='COLUMN:TYPE', multiple=True,
+              help='Cast parameter for each metadata column that should'
+              ' contain a specified column type (supported types can be'
+              ' found on docs.qiime2.org). The required formatting for this'
+              ' parameter is --cast COLUMN:TYPE, repeated for each column'
+              ' and the associated column type it should be cast to in'
+              ' the resultant output.')
+@click.option('--ignore-extra', is_flag=True,
+              help='Cast parameters that do not correspond to any of the'
+              ' column names within the provided metadata will result in'
+              ' a raised error unless this flag is enabled.')
+@click.option('--error-on-missing', is_flag=True,
+              help='If this flag is enabled, failing to include cast'
+              ' parameters for all columns in the provided metadata will'
+              ' result in an error.')
+@click.option('--output-file', required=True,
+              type=click.Path(exists=False, file_okay=True, dir_okay=False,
+                              writable=True),
+              help='Path to file or directory where the'
+              ' modified metdata should be written to.')
+@click.argument('paths', nargs=-1, required=True, metavar='METADATA...',
+                type=click.Path(exists=True, file_okay=True, dir_okay=False,
+                                readable=True))
+def cast_metadata(paths, cast, output_file, ignore_extra,
+                  error_on_missing):
+    import tempfile
+    import qiime2
+    import numpy as np
```

Comment:
I don't see a corresponding update listing `numpy` as a dependency in the [package recipe](https://github.com/qiime2/q2cli/blob/master/ci/recipe/meta.yaml), which is an issue in general when adding new package deps.

Bigger picture though, no need for `numpy` here for basic set operations, I'll explain below! Follow me...